ios/Classes/PTFlutterViewController.m

```diff
@@ -150,4 +154,60 @@ - (void)pdfViewCtrl:(PTPDFViewCtrl *)pdfViewCtrl downloadEventType:(PTDownloaded
     [super pdfViewCtrl:pdfViewCtrl downloadEventType:type pageNumber:pageNum message:message];
 }
 
+#pragma mark - Viewer Settings
+
+- (void)initViewerSettings
+{
+    [self setReadOnly:NO];
+    [self setThumbnailViewEditingEnabled:YES];
+}
+
+- (void)applyViewerSettings
+{
+    [self applyReadOnly];
+    
+    // Thumbnail editing enabled.
+    self.thumbnailsViewController.editingEnabled = self.thumbnailViewEditingEnabled;
+    
+    // Continuous annotation editing.
+    self.toolManager.tool.backToPanToolAfterUse = !self.continuousAnnotationEditing;
+    
+    // Annotation author.
+    self.toolManager.annotationAuthor = self.annotationAuthor;
+}
+
+- (void)applyReadOnly
+{
+    // Enable readonly flag on tool manager *only* when not already readonly.
+    // If the document is being streamed or converted, we don't want to accidentally allow editing by
+    // disabling the readonly flag.
+    if (![self.toolManager isReadonly]) {
+        self.toolManager.readonly = _readOnly;
+    }
+    
+    self.thumbnailsViewController.editingEnabled = !_readOnly;
+}
+
+- (void)setAnnotationAuthor:(NSString *)annotationAuthor {
+    _annotationAuthor = [annotationAuthor copy];
```

Comment:
The call to `copy` is unnecessary because `_annotationAuthor`'s storage type is `copy`. Unless there is a complier optimization here, this code will result in 2 copies.